### ISSUE TITLE
Official support for multi-instance WSL

### ISSUE THREAD

**Issue by long-nguyenxuan** [microsoft/WSL#5826]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I need to uninstall and re-install wsl for change development environement, because I use Windows as my daily-use and WSL as development-use 
I maintain some application with some conflicts (differenet JDK version, Python 2 vs Python 3 etc...) , but those all setup on Ubuntu 1804. 
Now when I want to update something on app01 (which use JDK8 and Python 2), I must uninstall current wsl (current setup with JDK 14 and Python 3) 

**Describe the solution you'd like**
Official support for multi-instance wsl. 

**Describe alternatives you've considered**
N/A 

**Additional context**
Currently, I'm doing some hack on registry to archive multi-instance-wsl, but an official support for multi-instance wsl is much better. 
ref: https://stackoverflow.com/questions/51584765/how-do-you-install-multiple-separate-instances-of-ubuntu-in-wsl

**Comment by Biswa96** [comment]
What's the problem with the answer in that Stackoverflow link? It already explains the import and export commands.

**Comment by sirredbeard** [comment]
You can have multiple instances of the same distro installed. You can manage them from PowerShell.

You can start with a base image you have configured, --export it, then --import it under a different name and now you have two.

You can have different Ubuntu releases installed, including 18.04 or 20.04, or --import a wsl image for all the older and interim releases from http://cloud-images.ubuntu.com.

https://wiki.ubuntu.com/WSL#Ubuntu_on_WSL
https://docs.microsoft.com/en-us/windows/wsl/wsl-config#managing-multiple-linux-distributions

**Comment by wanfuse123** [comment]
Hi!

I have exported a Ubuntu-18.04 guest with all the base system configuration I need

I have then imported this as a different distribution name

when I run it with
wsl --distribution Ubuntu-18.04-Project-A

It starts at the following prompt
/mnt/c/WINDOWS/system32

df -h shows
df: /proc/sys/fs/binfmt_misc: Too many levels of symbolic links
Filesystem Size Used Avail Use% Mounted on
/dev/sdd 251G 7.0G 232G 3% /
tools 1.9T 1.4T 528G 72% /init
none 2.9G 0 2.9G 0% /dev
tmpfs 2.9G 0 2.9G 0% /sys/fs/cgroup
none 2.9G 456K 2.9G 1% /run
none 2.9G 0 2.9G 0% /run/lock
none 2.9G 0 2.9G 0% /run/shm
none 2.9G 0 2.9G 0% /run/user
drivers 1.9T 1.4T 528G 72% /usr/lib/wsl/drivers
lib 1.9T 1.4T 528G 72% /usr/lib/wsl/lib
none 2.9G 0 2.9G 0% /mnt/wsl
/dev/sdc 251G 5.8G 233G 3% /mnt/wsl/docker-desktop-data/isocache
none 2.9G 12K 2.9G 1% /mnt/wsl/docker-desktop/shared-sockets/host-services
/dev/sdb 251G 122M 239G 1% /mnt/wsl/docker-desktop/docker-desktop-proxy
/dev/loop0 355M 355M 0 100% /mnt/wsl/docker-desktop/cli-tools
drvfsa 1.9T 1.4T 528G 72% /mnt/c
drvfsa 549M 34M 516M 7% /mnt/d
drvfsa 920G 484G 437G 53% /mnt/h

I am struggling to understand how further configuration of this new "Application Level VM" is kept separate from other copies of Ubuntu-18.04 and from the windows host system and from other linux system guests.

I was expecting it to be contained in a vhd file, but this does not appear to be the case!

A detailed explination would be greatly appreciated since I have a lot ridding on getting this to work with NVIDIA/CUDA/DOCKER for different projects that I want to spin up and spin down several times a day and keep as isolated from one another as much as possible.

Also is it possible to use NVIDIA in more than one container at a time or do I have to shutdown one and then start the next?

**Comment by cesarvarela** [comment]
Yeah, I'm confused about the expected use of WSL? Why is there no *easy*  copy/delete/rename?

`wsl --copy ubuntu ubuntu2`

`wsl --delete ubuntu2` 

`wsl --rename ubuntu2 dev-ubuntu`

**Comment by longtth** [comment]
> Yeah, I'm confused about the expected use of WSL? Why is there no _easy_ copy/delete/rename?
> 
> `wsl --copy ubuntu ubuntu2`
> 
> `wsl --delete ubuntu2`
> 
> `wsl --rename ubuntu2 dev-ubuntu`

yes, this is what I need
